api/src/main/java/org/openmrs/module/attachments/ComplexObsSaver.java

```diff
@@ -64,10 +65,23 @@ protected void prepareComplexObs(Visit visit, Person person, Encounter encounter
 		obs.setComment(fileCaption);
 	}
 	
+	/**
+	 * @deprecated as of 2.5.0, use
+	 *             {@link #saveImageAttachment(Visit, Person, Encounter, Concept, String, MultipartFile, String)}
+	 */
 	public Obs saveImageAttachment(Visit visit, Person person, Encounter encounter, String fileCaption,
 	        MultipartFile multipartFile, String instructions) throws IOException {
-		
-		conceptComplex = context.getConceptComplex(ContentFamily.IMAGE);
+		return saveImageAttachment(visit, person, encounter, null, fileCaption, multipartFile, instructions);
```

Comment:
Why deprecate it? Just keep it and make sure it invokes the new signature as such:
```java
conceptComplex = context.getConceptComplex(ContentFamily.IMAGE);
return saveImageAttachment(visit, person, encounter, conceptComplex, fileCaption, multipartFile, instructions);
```

Reply:
Resolved